js/private/repo_toolchains.bzl

```diff
@@ -0,0 +1,37 @@
+"Helpers to locate toolchains from within repository rules"
+
+load("@aspect_bazel_lib//lib:repo_utils.bzl", "repo_utils")
+
+def yq_path(rctx):
+    """Find the path to the host resolved yq binary from a repository rule.
+
+    Args:
+        rctx: The repository rule context
+
+    Returns:
+        Path to yq
+    """
+    return rctx.path(Label("{yq}{maybe_ext}".format(
```

Comment:
You can use
```
result = rctx.path(rctx.attr.yq)
```
without issue on Windows even if that files is not valid and then append .exe to `result`

Reply:
It looks like you can't add a string to a Path.

Reply:
right, rctx.path returns a Path object. cool cool